File: app/src/main/java/com/im_hero/diffpatch/BsDiff.java
package com.im_hero.diffpatch;

/**
 * Created on 2018/8/15.
 *
 * @author Jason
 * @version 1.0
 */
public class BsDiff {

    static {
        System.loadLibrary("bsdiff");
    }

    public native static int bsdiff(String oldFile, String newFile, String patchFile);

    public native static int bspatch(String oldFile, String newFile, String patchFile);
}

File: app/src/main/java/com/im_hero/diffpatch/BsDiff.kt
package com.im_hero.diffpatch

/**
 * Created on 2018/8/15.
 *
 * @author Jason
 * @version 1.0
 */
object BsDiff {

    init {
        System.loadLibrary("bsdiff")
    }

    external fun bsdiff(oldFile: String, newFile: String, patchFile: String): Int

    external fun bspatch(oldFile: String, newFile: String, patchFile: String): Int
}
